From 32721d3e5566cdd214d5e85109e5e5ccd2592a98 Mon Sep 17 00:00:00 2001
From: He Chuyue <hechuyue@wxiat.com>
Date: Wed, 24 Aug 2022 09:56:54 +0800
Subject: [PATCH] sw64: perf: fix perf_get_regs_user
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 32721d3e5566cdd214d5e85109e5e5ccd2592a98
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I56X48

--------------------------------

The past implementation of perf_get_regs_user could not get regs by
default, but it was actually available via task_pt_regs. Fix it now.

Signed-off-by: He Chuyue <hechuyue@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/perf_regs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/sw_64/kernel/perf_regs.c b/arch/sw_64/kernel/perf_regs.c
index 4c12a2cdf912..b036f213936b 100644
--- a/arch/sw_64/kernel/perf_regs.c
+++ b/arch/sw_64/kernel/perf_regs.c
@@ -28,6 +28,6 @@ u64 perf_reg_abi(struct task_struct *task)
 void perf_get_regs_user(struct perf_regs *regs_user,
 			struct pt_regs *regs)
 {
-	regs_user->regs = NULL;
-	regs_user->abi = PERF_SAMPLE_REGS_ABI_NONE;
+	regs_user->regs = task_pt_regs(current);
+	regs_user->abi = perf_reg_abi(current);
 }
-- 
2.33.0

